### ISSUE TITLE
Gran Turismo 6 crash after ending a race

### ISSUE THREAD

**Issue by xddxd** [RPCS3/rpcs3#8333]
In Gran Turismo 6, when I finish a race in the "Grand Valley speedway" track the game crashes with the following error: `SIG: Thread terminated due to fatal error: [index buffer] Working buffer not big enough, buffer_length=16777216 allocated=2084608 requested=4294967290 guard=65536 largest_pool=39084
(in file d:\a\1\s\rpcs3\Emu\RSX\Common\ring_buffer_helper.h:91)`
I do not know if this is regression or not, since I've never been able to finish a race in this track without desynchronizing before.
Log - [RPCS3.zip](https://github.com/RPCS3/rpcs3/files/4724309/RPCS3.zip)
Specs:
R7 2700
RX 580 20.5.1
Windows 10 2004

**Comment by mignolo4** [comment]
It also happen here: https://github.com/RPCS3/rpcs3/issues/8300#issue-626950267
maybe related?

**Comment by kd-11** [comment]
Needs checking if this is a regression

**Comment by xddxd** [comment]
If it is a regression, then it's a very old one. I went as far back as 0.0.7-8871 and the issue is still present.

**Comment by mignolo4** [comment]
My issue is slighty different because I cannot START  the race on that track (Grand Valley Speedway); I have the same problem entering the Kart Space track but I'm sure I drove in that track in the recent past.

Btw it doesn't happen with opengl (I had other issue but not the "Working buffer not big enough" and after a couple of tries I managed to enter the race @6 fps :-(

**Comment by Bertanos** [comment]
The "Disable FIFO reordering" option doesn't trigger this error

**Comment by mignolo4** [comment]
Trying to be helpful I tested all the versions for a regression and discovered:

**after #6571** (rpcs3-v0.0.7-8741-e0005ec3_win64) if I enter either the Grand Valley Speedway or the Kart Space Track I or II, I get:
`F {RSX [0x0bbcf9c]} RSX: class std::runtime_error thrown: [index buffer] Working buffer not big enough, buffer_length=67108864 allocated=1726664 requested=4294967290 guard=65536 largest_pool=67656
(in file C:\projects\rpcs3\rpcs3\Emu\RSX\Common\ring_buffer_helper.h:80)`

after a couple of seconds the starting grid shows up;

With one of the **last build** I get something similar but not identical:
`F {RSX [0x0bbcf9c]} SIG: Thread terminated due to fatal error: [index buffer] Working buffer not big enough, buffer_length=16777216 allocated=9728 requested=4294967290 guard=65536 largest_pool=512
(in file D:\a\1\s\rpcs3\Emu\RSX\Common\ring_buffer_helper.h:91)`

No problem at all with the previous version , so with rpcs3-v0.0.7-8739-c50119ae_win64 and older.

**Comment by MSuih** [comment]
Closing this as it seems to be a duplicate of #7913
